eclipse-collections-code-generator/src/main/resources/test/set/mutable/abstractImmutablePrimitiveSetTestCase.stg

```diff
@@ -263,6 +263,16 @@ public abstract class AbstractImmutable<name>HashSetTestCase extends AbstractImm
         Assert.assertEquals(set.toSet(), set.asLazy().toSet());
         Verify.assertInstanceOf(Lazy<name>Iterable.class, set.asLazy());
     }
+
+    @Test
+    public void union()
```

Comment:
I think we should add tests for both sets being empty, and for either being empty.

Reply:
Added tests for 1. both sets being empty. 2. left (set1) is empty 3. right (set2) is empty. 
Lines: 285-298.